opennebula/resource_opennebula_virtual_machine.go

```diff
@@ -470,6 +596,61 @@ func flattenVMDisk(d *schema.ResourceData, vmTemplate *vm.Template) error {
 	return nil
 }
 
+func flattendNICComputed(nic shared.NIC) map[string]interface{} {
+	nicID, _ := nic.ID()
+	sg := make([]int, 0)
+	ip, _ := nic.Get(shared.IP)
+	mac, _ := nic.Get(shared.MAC)
+	physicalDevice, _ := nic.GetStr("PHYDEV")
+	network, _ := nic.Get(shared.Network)
+
+	model, _ := nic.Get(shared.Model)
+	securityGroupsArray, _ := nic.Get(shared.SecurityGroups)
+
+	sgString := strings.Split(securityGroupsArray, ",")
+	for _, s := range sgString {
+		sgInt, _ := strconv.ParseInt(s, 10, 32)
+		sg = append(sg, int(sgInt))
+	}
+
+	return map[string]interface{}{
+		"nic_id":                   nicID,
+		"network":                  network,
+		"computed_ip":              ip,
+		"computed_mac":             mac,
+		"computed_physical_device": physicalDevice,
+		"computed_model":           model,
+		"computed_security_groups": sg,
+	}
+}
+
+// flattenVMTemplateNIC read NIC that come from template when instantiating a VM
+func flattenVMTemplateNIC(d *schema.ResourceData, vmTemplate *vm.Template) error {
+
+	// Set Nics to resource
+	nics := vmTemplate.GetNICs()
+	nicList := make([]interface{}, 0, len(nics))
+
+	for i, nic := range nics {
+
+		networkID, _ := nic.GetI(shared.NetworkID)
+		nicRead := flattendNICComputed(nic)
+		nicRead["network_id"] = networkID
+		nicList = append(nicList, nicRead)
+
+		if i == 0 {
+			d.Set("ip", nicRead["computed_ip"])
```

Comment:
Why forcing IP to the first one and not the other ones ?

Reply:
Mac adress is not managed too.

Reply:
There was an `ip` attribute at root of VM resource attribute added in initial provider code, I just copied the code to not change the behavior in this PR.
There is no MAC attribute a root.

Here is it's declaration:
```
"ip": {
			Type:        schema.TypeString,
			Computed:    true,
			Description: "Primary IP address assigned by OpenNebula",
	},
```

We may want to consider among some refactoring remove this attribute ?

Reply:
You are right. I made a mistake